src/Frame.js

```diff
@@ -0,0 +1,60 @@
+function Frame() {
+  // This tracks the rolls for a frame
+  this.scores = [null, null];
+  this.flatscore = 0;
+  this.ball = 0;
+  this.isOver = false;
+
+  this.bonusBalls = 0;
+  this.bonusscore = 0;
+  this.bonusAdded = false;
+};
+
+Frame.prototype.add_score = function(score) {
+  this.scores[this.ball] = score;
+  this.flatscore += score;
```

Comment:
Do you need this? Could you not instead have a getScore method which added the scores of the this.scores?